agent/worker.go

```diff
@@ -31,6 +32,9 @@ type Worker interface {
 	//
 	// The listener will be removed if the context is cancelled.
 	Listen(ctx context.Context, reporter StatusReporter)
+
+	// GetSecret returns a secret from the local cache
+	GetSecret(id string) (*api.Secret, error)
```

Comment:
It's kinda strange to add `GetSecret` to the `Worker` interface (which currently is `Init`, `Assign`, ... - very worker specific).

@stevvooe @cyli Thoughts?